### PR TITLE
fix(vite): use url for entry on windows

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves https://github.com/nuxt/framework/issues/6087

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This resolves the entry path via a file URL on Windows to skip vite trying to resolve it as an import from `nuxt3` directly.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/vite/src/vite.ts [hunk 1]
```diff
@@ -84,6 +86,10 @@ export async function bundle (nuxt: Nuxt) {
     )
   }
 
+  if (isWindows) {
+    ctx.config.resolve.alias['/entry.mjs'] = pathToFileURL(ctx.config.resolve.alias['/entry.mjs']).href
```
Comment (pi0) [inline packages/vite/src/vite.ts:90]: Any reason not doing this by default for all platforms?

File: packages/vite/src/dev-bundler.ts [hunk 2]
```diff
@@ -40,21 +40,17 @@ async function transformRequest (opts: TransformOptions, id: string) {
...
+  if (id && !id.startsWith('/@fs/') && id.startsWith('/')) {
     // Relative to the root directory
     const resolvedPath = resolve(opts.viteServer.config.root, '.' + id)
     if (existsSync(resolvedPath)) {
       id = resolvedPath
     }
   }
 
+  // On Windows, we prefix absolute paths with `/@fs/` to skip node resolution algorithm
+  id = id.replace(/^\/?(?=\w:)/, '/@fs/')
```
Comment (pi0) [inline packages/vite/src/dev-bundler.ts:52]: - Is there an upstream issue in vite to track or it is related to internal use?
- Can we consistantly add `/@fs` prefix always?

Comment (danielroe) [reply packages/vite/src/dev-bundler.ts:52]: We can always add `/@fs/` for absolute URLs, but some incoming requests _look_ like absolute URLs but are meant to be relative to the root directory, which is why I have only added it for drive letters/windows.

I'm not aware of an upstream vite issue. I can try to create reproduction but suspect the issue is our dev-bundler usage. (No fix was needed for vite-node.)

Comment (pi0) [reply packages/vite/src/dev-bundler.ts:52]: Okay thanks for explaining. Then let's keep as workaround and iterate over. We need to verify it on edge before release anyway :)

**Review by pi0 — APPROVED** [review]
<3

**Pull request merged**